client/extensions/woocommerce/app/products/product-form-variations-card.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FoldableCard from 'components/foldable-card';
+import ProductVariationTypesForm from './product-variation-types-form';
+import FormToggle from 'components/forms/form-toggle';
+
+export default class ProductFormVariationCard extends Component {
+
+	static propTypes = {
+		product: PropTypes.shape( {
+			id: PropTypes.isRequired,
+			type: PropTypes.string.isRequired,
+			name: PropTypes.string,
+		} ),
+		editProduct: PropTypes.func.isRequired,
+		editProductAttribute: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			isVariableProduct: props.product && 'variable' === props.product.type ? true : false,
+		};
+
+		this.handleToggle = this.handleToggle.bind( this );
+	}
+
+	handleToggle() {
+		this.setState( ( prevState ) => ( {
+			isVariableProduct: ! prevState.isVariableProduct,
```

Comment:
Why is this only updated in the internal state of the component? Shouldn't it be dispatched to the Redux tree? I'm assuming there are still a few pieces missing and this is one of them, but I just want to double-check.